core/io/resource_loader.cpp

```diff
@@ -420,6 +420,43 @@ Error ResourceLoader::load_threaded_request(const String &p_path, const String &
 	}
 }
 
+void ResourceLoader::load_threaded_cancel(const String &p_path) {
+	thread_load_mutex.lock();
+	if (!user_load_tokens.has(p_path)) {
+		thread_load_mutex.unlock();
+		return;
+	}
+
+	String local_path = user_load_tokens[p_path]->local_path;
+	if (thread_load_tasks.has(local_path)) {
+		ThreadLoadTask &load_task = thread_load_tasks.get(local_path);
+		while (load_task.status == THREAD_LOAD_IN_PROGRESS) {
+			if (load_task.cond_var) {
+				load_task.cond_var->notify_all();
+				memdelete(load_task.cond_var);
+				load_task.cond_var = nullptr;
+			}
+			thread_load_mutex.unlock();
+			OS::get_singleton()->delay_usec(1000);
+			thread_load_mutex.lock();
+		}
+		thread_load_tasks.erase(local_path);
+	}
+
+	memdelete(user_load_tokens[p_path]);
+	thread_load_mutex.unlock();
+}
+
+PackedStringArray ResourceLoader::get_requested_paths() {
+	PackedStringArray requested_paths;
+	thread_load_mutex.lock();
+	for (KeyValue<String, ResourceLoader::LoadToken *> &kvp : user_load_tokens) {
+		requested_paths.push_back(kvp.key);
+	}
+	thread_load_mutex.unlock();
```

Comment:
```suggestion
	MutexLock lock(thread_load_mutex);
	for (KeyValue<String, ResourceLoader::LoadToken *> &kvp : user_load_tokens) {
		requested_paths.push_back(kvp.key);
	}
```
Safer and preferred usage with trivial locking

Reply:
Thank you for reviewing my PR, I've applied your suggestion!